nursery/coin_select/src/coin_selector.rs

```diff
@@ -1,173 +1,171 @@
 use super::*;
+#[allow(unused)] // some bug in <= 1.48.0 sees this as unused when it isn't
+use crate::float::FloatExt;
+use crate::{bnb::BnBMetric, float::Ordf32, FeeRate};
+use alloc::{borrow::Cow, collections::BTreeSet, vec::Vec};
+
+/// [`CoinSelector`] is responsible for selecting and deselecting from a set of canididates.
+///
+/// You can do this manually by calling methods like [`select`] or automatically with methods like [`branch_and_bound`].
+///
+/// [`select`]: CoinSelector::select
+/// [`branch_and_bound`]: CoinSelector::branch_and_bound
+#[derive(Debug, Clone)]
+pub struct CoinSelector<'a> {
+    base_weight: u32,
+    candidates: &'a [Candidate],
+    selected: Cow<'a, BTreeSet<usize>>,
+    banned: Cow<'a, BTreeSet<usize>>,
+    candidate_order: Cow<'a, Vec<usize>>,
+}
 
-/// A [`WeightedValue`] represents an input candidate for [`CoinSelector`]. This can either be a
-/// single UTXO, or a group of UTXOs that should be spent together.
+/// A target value to select for along with feerate constraints.
 #[derive(Debug, Clone, Copy)]
-pub struct WeightedValue {
-    /// Total value of the UTXO(s) that this [`WeightedValue`] represents.
+pub struct Target {
+    /// The minimum feerate that the selection must have
+    pub feerate: FeeRate,
+    /// The minimum fee the selection must have
+    pub min_fee: u64,
```

Comment:
Since this only needs be specified for RBF, how about making it `Optional`?

Reply:
This would mean `Some(0)` and `None` have the same semantics which I think outweighs the better intuition provided by having it as an `Option`. Code docs should for sure explain that the only time you need to use this is to comply with RBF rules.